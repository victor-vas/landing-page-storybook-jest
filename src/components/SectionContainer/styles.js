import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    max-width: 120rem;
    width: 100%;
    margin: 0 auto;
    padding: ${theme.spacings.large};
  `}
`;
