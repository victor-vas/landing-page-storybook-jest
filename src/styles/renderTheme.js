import { ThemeProvider } from 'styled-components';
import { render } from '@testing-library/react';
import { theme } from './theme';

export const renderTheme = (children) => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider>);
};
