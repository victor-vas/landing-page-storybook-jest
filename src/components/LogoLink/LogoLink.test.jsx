import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/renderTheme';
import { LogoLink } from '.';

describe('<LogoLink />', () => {
  it('should render text logo', () => {
    renderTheme(<LogoLink link="#target" text="test" />);

    expect(screen.getByRole('link', { name: 'test' })).toHaveAttribute(
      'href',
      '#target',
    );
  });

  it('should render image logo', () => {
    renderTheme(<LogoLink link="#target" text="test" srcImg="image.png" />);

    expect(screen.getByRole('img', { name: 'test' })).toHaveAttribute(
      'src',
      'image.png',
    );
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(
      <LogoLink link="#target" text="test" srcImg="image.png" />,
    );

    expect(container).toMatchSnapshot();
  });
});
