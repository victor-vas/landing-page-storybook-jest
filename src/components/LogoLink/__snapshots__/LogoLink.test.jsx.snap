// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogoLink /> should match snapshot 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c1 > img {
  max-height: 3rem;
}

.c0 {
  color: #0A1128;
  font-size: 6.4rem;
  text-transform: none;
}

@media (max-width:768px) {
  .c0 {
    font-size: 4.0rem;
  }
}

<div>
  <h1
    class="c0"
  >
    <a
      class="c1"
      href="#target"
    >
      <img
        alt="test"
        src="image.png"
      />
    </a>
  </h1>
</div>
`;
